modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/components/rules/editor/Editor.es.js

```diff
@@ -129,8 +131,21 @@ const reducer = (state, action) => {
 			const {actions, conditions} = state.ifStatement;
 			const {loc, value, ...otherPayloads} = action.payload;
 
+			let newActions = actions[loc];
+
+			if (
+				actions[loc].action === 'auto-fill' &&
+				actions[loc].label !== value
+			) {
+				newActions = {
+					...actions[loc],
+					inputs: {},
+					outputs: {},
+				};
+			}
```

Comment:
Use `newActions` instead of `actions[loc]`
```suggestion
			if (newActions === 'auto-fill' && newActionsl !== value) {
				newActions = {
					...newActions,
					inputs: {},
					outputs: {},
				};
			}
```